Use ConsoleCliDownloads instead of hardcoded cli tools

jhadvig opened the pull request:
Replacing hardcoded cli tools for ConsoleCliDownloads that are handled by console-operator, addressed by  https://github.com/openshift/console-operator/pull/306

This leaves an open question how to display empty state of the `Command Line Tools` page.

@spadgett @openshift/team-ux-review any suggestions ?

benjaminapetersen commented:
lgtm pondering before or after the operator change merges.

jhadvig commented:
@spadgett you mean `Command Line Tools` item in the app luncher ?

<img width="331" alt="Screen Shot 2019-10-11 at 13 14 17" src="https://user-images.githubusercontent.com/1668218/66647436-36e06280-ec29-11e9-90f1-1f257ad910e9.png">

spadgett commented:
> @spadgett you mean `Command Line Tools` item in the app luncher ?

In the help menu? Yes. It will always be empty if the CLI download CRD doesn't exist, so we should hide it.

jhadvig commented:
@spadgett I've update the PR. No the `Command Line Tools` item in the Help menu will only appear if there is any ConsoleCLIDownloads CR.

PTAL

jhadvig commented:
/retest

jhadvig commented:
-retest

jhadvig commented:
/retest

spadgett commented on frontend/public/actions/features.ts:
```diff
@@ -154,6 +160,18 @@ const detectConsoleLinks = (dispatch) =>
     },
   );
 
+const detectConsoleCLIDownloads = (dispatch) =>
```
I would check for the presence of the CRD rather than making the request.

jhadvig commented:
@spadgett comment addressed. PTAL

spadgett commented on frontend/public/components/masthead-toolbar.jsx:
```diff
@@ -284,25 +284,33 @@ class MastheadToolbar_ extends React.Component {
+    ];
+
+    if (!flags[FLAGS.CONSOLE_CLI_DOWNLOAD]) {
+      _.remove(actions, (a) => a.label === 'Command Line Tools');
```
```js
    const actions = [
      {
        label: 'Documentation',
        externalLink: true,
        href: openshiftHelpBase,
      },
      ...flags[FLAGS.CONSOLE_CLI_DOWNLOAD] ? [{
        label: 'Command Line Tools',
        callback: this._onCommandLineTools,
      }] : [],
      {
        label: 'About',
        callback: this._onAboutModal,
      },
    ];

jhadvig commented:
@spadgett I've updated the PR with addressing your comments. 
Placed the request token as a first item in the "Command Line Tools" page. 
Screen:
<img width="777" alt="Screen Shot 2019-10-21 at 11 28 36" src="https://user-images.githubusercontent.com/1668218/67196181-99d8b300-f3fa-11e9-99fa-686b32e11069.png">

Wasn't really sure if any additional header/text is needed since the link text is pretty descriptive itself.

PTAL

jhadvig commented:
/retest

spadgett commented:
@beanh66 any thoughts on where the copy login command link should go on this page?

spadgett commented on frontend/public/components/command-line-tools.tsx:
```diff
@@ -54,44 +54,15 @@ const CommandLineTools: React.FC<CommandLineToolsProps> = ({ obj }) => {
-          <ExternalLink href={ODO_DOWNLOAD_LINK} text="Download odo" />
-        </p>
+        {(window as any).SERVER_FLAGS.requestTokenURL && (
+          <React.Fragment>
```
nit: Prefer the `<>` fragment shorthand

```suggestion
          <>
```

spadgett commented on frontend/public/components/command-line-tools.tsx:
```diff
@@ -54,44 +54,15 @@ const CommandLineTools: React.FC<CommandLineToolsProps> = ({ obj }) => {
+              href={(window as any).SERVER_FLAGS.requestTokenURL}
+              text="Copy Login Command"
+            />
+          </React.Fragment>
```
```suggestion
          </>
```

spadgett commented on frontend/public/components/masthead-toolbar.jsx:
```diff
@@ -592,5 +598,10 @@ const mastheadToolbarStateToProps = ({ UI }) => ({
+    FLAGS.AUTH_ENABLED,
+    FLAGS.OPENSHIFT,
+    FLAGS.CLUSTER_VERSION,
+    FLAGS.CONSOLE_CLI_DOWNLOAD,
```
nit: I tend to alphabetize lists when they start to get large and order doesn't matter

spadgett commented on frontend/public/components/namespace.jsx:
```diff
@@ -285,7 +284,7 @@ const ProjectList_ = (props) => {
       </p>
       <p>
-        Download the <ExternalLink href={OC_DOWNLOAD_LINK} text="command-line tools" />.
+        Download the <Link to={`/command-line-tools`}>command-line tools</Link>
```
```suggestion
        Download the <Link to="/command-line-tools">command-line tools</Link>
```

jhadvig commented:
@spadgett PR updated.
@beanh66 can you please check https://github.com/openshift/console/pull/2951#issuecomment-544444243 and 
https://github.com/openshift/console/pull/2951#issuecomment-544592956
for review... Thanks !
